helm-app-operator/pkg/helm/helm.go

```diff
@@ -193,58 +187,71 @@ func getWatchesFile() (string, bool) {
 	return "", false
 }
 
-// InstallRelease accepts a custom resource, installs a Helm release using Tiller,
+// ReconcileRelease accepts a custom resource, ensures the described release is deployed,
 // and returns the custom resource with updated `status`.
-func (c installer) InstallRelease(r *unstructured.Unstructured) (*unstructured.Unstructured, error) {
+// - If the custom resource does not have a release, a new release will be installed
+// - If the custom resource has changes for an existing release, the release will be updated
+// - If the custom resource has no changes for an existing release, the underlying resources will be reconciled.
+func (c installer) ReconcileRelease(r *unstructured.Unstructured) (*unstructured.Unstructured, bool, error) {
```

Comment:
One more thing. Not that I can see this anywhere but you might want to double check that none of the installer methods are modifying the installer. Otherwise we should use pointer receivers `(c *installer)`. If not then this is fine.
Just pointing it out since its been a while since I've seen non pointer receivers.

Reply:
The struct fields are `storageBackend`, `tillerKubeClient` and `chartDir`, none of which should change, so I think non-pointer receivers is what we want.